server/src/main/java/org/elasticsearch/cluster/routing/package-info.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/**
+ * <p>Cluster routing package provides routing information and manages shard
+ * allocations in cluster. The routing part contains two different views on
+ * shards in cluster {@link org.elasticsearch.cluster.routing.RoutingTable} and
+ * {@link org.elasticsearch.cluster.routing.RoutingNodes}. RoutingTable provides
+ * view from index to shard to node, and RoutingNodes provides view from node to
+ * shard. Shard allocation is a process of assigning and moving shards between
+ * nodes. For more details about allocation see {@link
+ * org.elasticsearch.cluster.routing.allocation}.</p>
+ *
+ * <b>Routing Table</b>
+ *
+ * <p>The routing table provide a view on global cluster state from index
+ * perspective. It's a mapping from indices to shards and shards to nodes, where
+ * relationship between shard and node might not exists if shard allocation is
+ * not possible. Routing table represents desired shards layout and current
+ * state of shards.</p>
+ *
+ * <p>As class hierarchy goes - at the top level there is a {@link
+ * org.elasticsearch.cluster.routing.RoutingTable}. RoutingTable is a part of
+ * {@link org.elasticsearch.cluster.ClusterState}. It contains mapping between
+ * indices and {@link org.elasticsearch.cluster.routing.IndexRoutingTable}.
+ * {@link org.elasticsearch.cluster.routing.IndexRoutingTable} contains routing
+ * information about all shards in that index - {@link
+ * org.elasticsearch.cluster.routing.IndexShardRoutingTable}. Each shard can
+ * have one or more instances - primary and replicas. IndexShardRoutingTable
+ * contains information about all shard instances for specific shard id - {@link
+ * org.elasticsearch.cluster.routing.ShardRouting}.</p>
+ *
+ * <b>Routing Nodes</b>
+ *
+ * <p>{@link org.elasticsearch.cluster.routing.RoutingNode} provides view to
+ * shard routing from node perspective. There is a RoutingNode for every {@link
+ * org.elasticsearch.cluster.node.DiscoveryNode}, and it contains information about
```

Comment:
I don't see a RoutingNode object in DiscoveryNode? Is there something more to this relationship that could be said here?

Reply:
A RoutingNode has association with DiscoveryNode, not vice versa.